File: glenn/136/go/singleNumber.go
package glenn

func singleNumber(nums []int) int {
	h := make(map[int]int, 0)
	for _, num := range nums {
		if _, p := h[num]; p {
			delete(h, num)
		} else {
			h[num] = 1
		}
	}

	for k := range h { return k }
	return -1
}

// Runtime: 24 ms, faster than 31.91% of Go online submissions for Single Number.
// Memory Usage: 6.3 MB, less than 13.76% of Go online submissions for Single Number.

File: glenn/136/ruby/single_number.rb
# @param {Integer[]} nums
# @return {Integer}
def single_number(nums)
  instance_hash = {}
  nums.each do |num|
    if instance_hash[num].nil?
      instance_hash[num] = 1
    else
      instance_hash.delete(num)
    end
  end

  return nil unless instance_hash.keys.count == 1
  instance_hash.keys[0]
end


# Runtime: 48 ms, faster than 98.72% of Ruby online submissions for Single Number.
# Memory Usage: 212.8 MB, less than 28.63% of Ruby online submissions for Single Number.